src/status_im/ui/screens/network_settings/edit_network/events.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im.ui.screens.network-settings.edit-network.events
+  (:require [re-frame.core :as re-frame]
+            [status-im.utils.handlers :refer [register-handler] :as handlers]
+            [status-im.utils.handlers-macro :as handlers-macro]
+            [status-im.ui.screens.accounts.utils :as accounts.utils]
+            [status-im.utils.ethereum.core :as ethereum]
+            [status-im.utils.types :as types]
+            [clojure.string :as string]))
+
+(defn- new-network [{:keys [random-id] :as cofx} network-name upstream-url type]
+  (let [data-dir (str "/ethereum/" (name type) "_rpc")
+        config   {:NetworkId      (ethereum/chain-keyword->chain-id type)
+                  :DataDir        data-dir
+                  :UpstreamConfig {:Enabled true
+                                   :URL     upstream-url}}]
+    {:id         (string/replace random-id "-" "")
+     :name       network-name
+     :config     (types/clj->json config)
+     :raw-config config}))
+
+(handlers/register-handler-fx
+ :save-new-network
+ [(re-frame/inject-cofx :random-id)]
+ (fn [{{:networks/keys [manage] :account/keys [account] :as db} :db :as cofx} _]
+   (let [{:keys [name url chain]} manage
+         network                  (new-network cofx (:value name) (:value url) (:value chain))
+         new-networks             (merge {(:id network) network} (:networks account))]
+     (handlers-macro/merge-fx cofx
+                              {:db       (dissoc db :networks/manage)
+                               :dispatch [:navigate-back]}
+                              (accounts.utils/account-update {:networks new-networks})))))
+
+(handlers/register-handler-fx
+ :network-set-input
+ (fn [{db :db} [_ input-key value]]
+   {:db (update db :networks/manage merge {input-key {:value value
+                                                      :error (and (string? value) (empty? value))}})}))
```

Comment:
Just a side note: there is no validation for `UpstreamConfig.URL` field, but we assume that user using this feature is advanced enough to enter valid url.

Reply:
I have had bad luck in the past trying to validate urls in general since this url may be an ip or an address with a port or something more elaborate that a regex might not allow. Thus I think it would be better to allow any non empty user input but gracefully error if a connection cannot be made.